--- build/pre-copied/common/CLASSLIB/modules/luni/src/main/java/com/ibm/oti/vm/MsgHelp.java	2006-03-22 00:02:58.510272000 +0300
+++ build/pre-copied/common/CLASSLIB/modules/luni/src/main/java/com/ibm/oti/vm/MsgHelp.java	2006-03-22 17:14:50.563239000 +0300
@@ -107,8 +107,7 @@ public final class MsgHelp {
 			return (ResourceBundle) AccessController
 					.doPrivileged(new PrivilegedAction() {
 						public Object run() {
-							return ResourceBundle.getBundle(resource, locale,
-									loader);
+							return ResourceBundle.getBundle(resource, locale, loader != null ? loader : ClassLoader.getSystemClassLoader());
 						}
 					});
 		} catch (MissingResourceException e) {
