--- pre-copied/common/CLASSLIB/modules/luni/src/main/java/java/net/URLClassLoader.java	2006-03-13 05:09:17.000000000 +0300
+++ pre-copied/common/CLASSLIB/modules/luni/src/main/java/java/net/URLClassLoader.java	2006-03-22 23:19:47.129704000 +0300
@@ -1017,6 +1017,7 @@ public class URLClassLoader extends Secu
 									+ name.length()).append(filename).append(
 									name).toString();
 						}
+						filename = URLDecoder.decode(filename, "UTF-8");
 						File file = new File(filename);
 						// Don't throw exceptions for speed
 						if (file.exists()) {
